### PR TITLE
Fix CategoricalMLPPolicy

### PR DESCRIPTION
**Pull request by CatherineSue**
The problem is that CategoricalMLPPolicy._f_prob and
tf.Distribution.Categorical. It returns wrong shape array.

Fixes: #263

### CODE REVIEW

File: garage/tf/policies/categorical_mlp_policy.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def __init__(
             with tf.name_scope(self._prob_network_name):
                 prob_network_outputs = L.get_output(prob_network.output_layer)
             self._f_prob = tensor_utils.compile_function(
-                [prob_network.input_layer.input_var], [prob_network_outputs])
+                [prob_network.input_layer.input_var], prob_network_outputs)
```
Comment (CatherineSue) [inline garage/tf/policies/categorical_mlp_policy.py]: This problem also appears in garage/tf/policies/deterministic_mlp_policy.py. Should I change it?

Comment (zhanpenghe) [reply garage/tf/policies/categorical_mlp_policy.py]: Yes, please do that.

**Comment by ryanjulian** [comment]
Can you add a test which is broken before this change and fixed after?

**Comment by CatherineSue** [comment]
I added a test file into examples/tf.

**Comment by ryanjulian** [comment]
If you are fixing DeterministicMLPPolicy, can you add a unit test for that as well?

**Comment by CatherineSue** [comment]
I think the DeterministicMLPPolicy is supposed to be used in DDPG, right? But currently, because we need to use gym.spaces.Dict to use DDPG+HER and garage spaces don't have the wrapper for it, we can't use TfEnv() wrapper when using DDPG. But DeterministicMLPPolicy is designed to use TfEnv() and the garage.spaces. I can change it to make DeterminisitcMLPPolicy work but I don't think it is a good idea.

**Comment by ryanjulian** [comment]
Okay, please file a bug and fix it with Jonathon in the next CL

**Pull request merged**